torch/nn/utils/stateless.py

```diff
@@ -37,21 +36,52 @@ def _setattr(self, name: str, value: Any) -> None:
     module._orig_class = cls
 
 
-def _check_tied_val_already_replaced(old_val, new_val, replaced_tensors_map):
-    if old_val not in replaced_tensors_map:
-        replaced_tensors_map[old_val] = new_val
-    elif replaced_tensors_map[old_val] is not new_val:
-        warnings.warn("functional_call was passed multiple values for tied weights. "
-                      "This behavior is deprecated and will be an error in future versions")
+def _create_tied_weights_map(module, params_and_buffers):
+    # creates a weight map of {tied_name: name_given_by_user} for all weights where one of their tied weights is passed
```

Comment:
nit: a simple example of what the weight map looks like might help, I remember reading through this some time ago, working through it, understanding it, and now I'm trying to re-remember how it works.

Reply:
Added